datasync/chaindatafetcher/kafka/kafka_test.go

```diff
@@ -119,6 +161,123 @@ func (s *KafkaSuite) TestKafka_Publish() {
 	}
 }
 
+func (s *KafkaSuite) TestKafka_Subscribe() {
+	numTests := 10
+	testBytesSize := 100
+
+	topic := "test-subscribe"
+	s.kfk.CreateTopic(topic)
+
+	expected := s.publishRandomData(topic, numTests, testBytesSize)
+
+	var actual []*kafkaData
+	s.subscribeData(topic, "test-group-id", numTests, func(message *sarama.ConsumerMessage) error {
+		var d *kafkaData
+		json.Unmarshal(message.Value, &d)
+		actual = append(actual, d)
+		return nil
+	})
+
+	// compare the results with the published data
+	s.Equal(expected, actual)
+}
+
+func (s *KafkaSuite) TestKafka_PubSubWith2Partitions() {
+	numTests := 10
+	testBytesSize := 100
+
+	s.kfk.config.Partitions = 2
+	defer func() { s.kfk.config.Partitions = DefaultPartitions }()
+
+	topicPartition := "test-2-partition-topic"
+	s.kfk.CreateTopic(topicPartition)
+
+	// publish random data
+	expected := s.publishRandomData(topicPartition, numTests, testBytesSize)
+
+	var actual []*kafkaData
+	s.subscribeData(topicPartition, "test-group-id", numTests, func(message *sarama.ConsumerMessage) error {
+		var d *kafkaData
+		json.Unmarshal(message.Value, &d)
+		actual = append(actual, d)
+		return nil
+	})
+
+	// the number of partitions is not 1, so the order may be changed after subscription.
+	// compare the results with the published data
+	s.Equal(len(expected), len(actual))
+
+	for _, e := range expected {
```

Comment:
Can't we compare them directly like `s.Equal(expected, actual)`?

Reply:
We can't. This is because the order of consuming is not assurable since the number of partitions is 2.